### DISCUSSION TITLE
Web service times out with default setting for "suspend execution for cached apps" (developer option, separate to power settings).

### DISCUSSION THREAD

**Discussion by tom-a-horrocks** [NightscoutFoundation/xDrip#3275] [Bug reports]
I have discovered ([troubleshooting here](https://github.com/swissalpine/GARMIN-xDrip-Spike-Nightscout-Watchface/issues/10)) that requests to the xDrip Web Service will time out if xDrip+ hasn't been in the foreground recently. I have disabled battery optimisations for Dexcom app, Xdrip+ app, system bluetooth app, and have also disabled adaptive battery. I've only been able to reproduce this problem on a Pixel 7 (Android 14); the web service worked fine on a Cubot KingKong Mini 3 (Android 12).

**Steps to reproduce**
1. Leave xDrip+ in the background for up to 2 hours.
4. Visit the test link (http://127.0.0.1:17580/sgv.json?count=12) and observe it never loads.
5. Open xDrip+ into the foreground.
6. Visit the test link again, and observe it _does_ load this time.

**Configuration**
* Phone: Pixel 7 
* Android 14
* xDrip+: 1d84ed1-2023.12.23, but also a problem with earlier versions.

**UPDATE**
It turns out Pixel 7 suspends the execution of cached (background) apps, which was preventing the web service from responding. This can be disabled globally by changing the developer option below. Maybe there's a way to fix this in the app that doesn't require changing a global setting?

> Settings -> System -> Developer options -> Suspend execution for cached apps -> "Disabled" (previously: "Device default")

That _mostly_ fixed the problem, but it was still a little flakey. I found that resetting my launcher helped with this... maybe the Android 13 to 14 update was not totally clean.  To reset the launcher, do the following. Warning: this resets your homescreen apps and widgets to factory, so you'll have to make all your shortcuts etc again.

> Settings -> Apps -> See all Apps -> <Three dot menu -> Show system> -> Pixel Launcher -> Storage and cache -> Clear storage

**Comment by damien640** [comment]
OMG I was looking for a solution to this problem for one year... thank you SO MUCH!!!

**Reply by tom-a-horrocks** [reply]
Glad it helped you too!

**Reply by jamorham** [reply]
Which collector are you using? @tom-a-horrocks

**Reply by tom-a-horrocks** [reply]
@jamorham 640G / EverSense (reading from the Build Your Own Dexcom App).

**Reply by damien640** [reply]
Same for me.
